test/runtime/fqdn.go

```diff
@@ -622,6 +621,81 @@ var _ = Describe("RuntimeFQDNPolicies", func() {
 		res = vm.ContainerExec(helpers.App2, helpers.CurlFail(target))
 		res.ExpectSuccess("Container %q cannot access to %q when should work", helpers.App2, target)
 	})
+
+	Context("FQDNS withtout poller", func() {
+		var config = `
+PATH=/usr/lib/llvm-3.8/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin
+CILIUM_OPTS=--kvstore consul --kvstore-opt consul.address=127.0.0.1:8500 --debug --pprof=true --log-system-load --tofqdns-enable-poller=false
+INITSYSTEM=SYSTEMD`
+		BeforeAll(func() {
+			vm.SetUpCiliumWithOptions(config)
+
+			ExpectCiliumReady(vm)
+			areEndpointsReady := vm.WaitEndpointsReady()
+			Expect(areEndpointsReady).Should(BeTrue(), "Endpoints are not ready after timeout")
+		})
+
+		AfterAll(func() {
+			vm.SetUpCilium()
+			_ = vm.WaitEndpointsReady() // Don't assert because don't want to block all AfterAll.
+		})
+
+		It("Policy addition after DNS lookup", func() {
+			policy := `
+[
+	{
+		"labels": [{
+			"key": "Policy addition after DNS lookup"
+		}],
+		"endpointSelector": {
+			"matchLabels": {
+				"container:id.app1": ""
+			}
+		},
+		"egress": [
+			{
+				"toPorts": [{
+					"ports":[{"port": "53", "protocol": "ANY"}],
+					"rules": {
+						"dns": [
+							{"matchName": "world1.cilium.test"},
+							{"matchPattern": "*.cilium.test"}
+						]
+					}
+				}]
+			},
+			{
+				"toFQDNs": [
+					{"matchName": "world1.cilium.test"},
+					{"matchPattern": "*.cilium.test"}
+				]
+			}
+		]
+	}
+]`
+
+			By("Testing connectivity to %q", world1Target)
+			res := vm.ContainerExec(helpers.App1, helpers.CurlFail(world1Target))
+			res.ExpectSuccess("Cannot access to %q", world1Target)
+
+			By("Importing the policy")
+			_, err := vm.PolicyRenderAndImport(policy)
+			Expect(err).To(BeNil(), "Policy cannot be imported")
+
+			By("Trying curl connection to %q without DNS request", world1Target)
+
+			res = vm.ContainerExec(helpers.App1, helpers.CurlFail(fmt.Sprintf(
+				"--resolve %s:%s", world1Target, worldIps[WorldHttpd1])))
+			res.ExpectFail(
```

Comment:
Confirming this logic - the reason this fails is because we specify not to poll for the IPs that `world1.cilium.test` resolves to because the poller is disabled, and we tell curl here not to resolve the IP of `world1Target`, which causes the lookup for the DNS proxy to fail? Is that correct? cc: @raybejjani

Reply:
Roughly, yes. The poller doesn't run (specified above) but curl also doesn't do a DNS lookup because `--resolve` gives it the IP. Since no DNS lookup happened, no IP will be placed in a generated toCIDRSet rule. As your other comment below points out, when a DNS lookup occurs then we do expect the IP to show up, and the request to succeed.